### ISSUE TITLE
Update ManureManagerConfigHandler getter function args with enum members instead of strings

### ISSUE THREAD

**Issue by ew3361zh** [RuminantFarmSystems/RuFaS#1118]
In MASM > RUFAS > routines > manure > IO_helpers > manure_manager_config_handler.py, update the arg for each of the getter functions with members from the appropriate enum classes.

This includes these four methods within the ManureManagerConfigHandler class:
`get_custom_bedding_config()`
`get_custom_manure_handler_config()`
`get_custom_manure_separator_config()`
`get_custom_manure_treatment_config()`

**Comment by ew3361zh** [comment]
Consider making manure enums in single file or if that doesn't seem like a good solution create a folder within the manure routines to contain all enums for the manure class.

**Comment by Isgotkowitz** [comment]
After working to address #1127 in PR #1250, I think this would be a bad change to make. Making the arg for these getters into an enum would severely reduce the configurability that the manure module currently has.

**Comment by Isgotkowitz** [comment]
@Isgotkowitz come back here and explain the above comment, and what should happen with this issue.

**Comment by Isgotkowitz** [comment]
In the methods listed above (except `get_custom_manure_handler_config()` which no longer exists), the string argument is converted to an Enum element and that element is used to find the appropriate configuration. This PR requests that an Enum element is passed instead of a string, but I think a better approach would be changing the configurations to be referenced with strings instead of Enum elements.

Manure Handler configs are looked up this way, but the string passed as an argument is the _name_ of the configuration that should be used, not the _type_. This is an important distinction because it allows multiple handlers of the same type to be used in a single simulation.

So instead of executing this issue, I propose the following steps be taken for the beddings, separators, and treatments:
- Add a "name" field to each configuration in the input.
- Enforce uniqueness of the "name" field when parsing inputs (i.e. when calling `_process_bedding_configs()`, `_process_manure_separator_configs()`, `_process_manure_treatment_configs()`).
- Remove the default configuration factories (this is what #1127 was about, it was just specific to manure handlers).
- Change the getter functions listed in the top comment to lookup configurations by name instead of type.

In short, the RuFaS should handle beddings, separators and treatments should be handled the same as handlers.

**Comment by Isgotkowitz** [comment]
This can be folded into #1237

**Comment by allisterakun** [comment]
Ed will write a new issue, closing this as unplanned.
